src/mirth/c99.mth

```diff
@@ -413,9 +413,21 @@ def(c99-args-op!, List(Arg) Op +C99 -- +C99,
     OP_COERCE   -> drop2,
     OP_LABEL_PUSH -> nip c99-label-push!,
     OP_LABEL_POP  -> nip c99-label-pop!,
+    OP_DATA_GET_TAG -> c99-get-data-tag!,
     OP_DATA_GET_LABEL -> c99-tag-get-label! drop,
     OP_DATA_SET_LABEL -> c99-tag-set-label! drop)
 
+def(c99-get-data-tag!, List(Arg) Type +C99 -- +C99,
+    drop2
+    c99-line("{" put)
+    c99-nest(
+        c99-line("VAL val = pop_value();" put)
+	c99-line("USIZE tag = get_data_tag(val);" put)
+	c99-line("push_u64(tag);" put )
+    )
+    c99-line("}" put)
+)
```

Comment:
A decref is needed. Also for enums this doesn't have to do anything, so I suggest:

```suggestion
def(c99-get-data-tag!, List(Arg) Data +C99 -- +C99,
    nip is-enum? else(
        c99-line("{" put)
        c99-nest(
            c99-line("VAL val = pop_value();" put)
	    c99-line("USIZE tag = get_data_tag(val);" put)
	    c99-line("push_u64(tag);" put)
	    c99-line("decref(val);" put)
        )
        c99-line("}" put)
    ))
```

Also for transparent data maybe it should always return 0 (or whatever the tag's value is), instead of calling `get_data_tag`.🤔